### PR TITLE
[fix](statistics)Release StmtExecutor in AnalysisTask object when sql execution finished to release memory.

### PR DESCRIPTION
**Pull request by Jibing-Li**
Release useless StmtExecutor reference in time to reduce memory useage.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Jibing-Li** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java [hunk 1]
```diff
@@ -183,14 +184,22 @@ protected ResultRow collectBasicStat(AutoCloseConnectContext context) {
...
-        stmtExecutor = new StmtExecutor(context.connectContext, stringSubstitutor.replace(BASIC_STATS_TEMPLATE));
-        return stmtExecutor.executeInternalQuery().get(0);
+        String sql = stringSubstitutor.replace(BASIC_STATS_TEMPLATE);
+        stmtExecutor = new StmtExecutor(context.connectContext, sql);
+        ResultRow resultRow = stmtExecutor.executeInternalQuery().get(0);
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Cost time in millisec: " + (System.currentTimeMillis() - startTime)
+                    + " Min max SQL: " + sql + " QueryId: " + DebugUtil.printId(stmtExecutor.getContext().queryId()));
+        }
+        stmtExecutor = null;
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java:201]: add some comment to explain why need set to null

Comment (Jibing-Li) [reply fe/fe-core/src/main/java/org/apache/doris/statistics/OlapAnalysisTask.java:202]: added

**Comment by Jibing-Li** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Review by morrySnow — APPROVED** [review]

**Pull request merged**